src/Sparc.jjt

```diff
@@ -1433,6 +1436,83 @@ SimpleNode head() :
     jjtThis.beginColumn = n.beginColumn;
     return jjtThis;
   }
+| n = optimize_statement()
+  {
+    jjtThis.beginLine = n.beginLine;
+    jjtThis.beginColumn = n.beginColumn;
+    return jjtThis;
+  }
+}
+        
+SimpleNode optimize_statement() :
+{
+  Token t;
+}
+{
+  t = < MINIMIZE_OB > [ optimizeParameterList() ] < COLON > [ nonRelAtomList() ] < CB >

```

Comment:
This could be implemented slightly cleaner if you introduce a new Token, something like:

Token optimize_ob() #void :
{
  Token t;
}
{
  t = <  MINIMIZE_OB  >
  {
  return t;
| t = < MAXIMIZE_OB  >
  {
    return t;
  }

and then use it:

t = optimize_ob() [ optimizeParameterList() ] < COLON > [ nonRelAtomList() ] < CB >
  { 
    jjtThis.image = t.image;
    jjtThis.beginLine = t.beginLine;
    jjtThis.beginColumn = t.beginColumn;
    if (Settings.getSingletonInstance().getSolver() != ASPSolver.Clingo) 
        throw new ParseException("line " + jjtThis.beginLine + ", column " + jjtThis.beginColumn + ": optimization statements are not supported in DLV");
    return jjtThis;
  }

In this way the code on lines 1454-1460 doesn't need to be repeated twice.

Reply:
Changed the code to include this change.